packages/react/menu/src/Menu.tsx

```diff
@@ -94,65 +237,179 @@ const MenuContent = React.forwardRef((props, forwardedRef) => {
-     */
-    trapFocus?: FocusScopeOwnProps['trapped'];
+type MenuRootContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  keyof MenuContentImplPrivateProps
+>;
+type MenuRootContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuRootContentOwnProps
+>;
 
-    /**
-     * Event handler called when auto-focusing on open.
-     * Can be prevented.
-     */
-    onOpenAutoFocus?: FocusScopeOwnProps['onMountAutoFocus'];
+const MenuRootContent = React.forwardRef((props, forwardedRef) => {
+  const context = useMenuContext(CONTENT_NAME);
+  const ref = React.useRef<React.ElementRef<typeof MenuContentImpl>>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
 
-    /**
-     * Event handler called when auto-focusing on close.
-     * Can be prevented.
-     */
-    onCloseAutoFocus?: FocusScopeOwnProps['onUnmountAutoFocus'];
+  // Hide everything from ARIA except the `MenuContent`
+  React.useEffect(() => {
+    const content = ref.current;
+    if (content) return hideOthers(content);
+  }, []);
 
-    /**
-     * Whether scrolling outside the `MenuContent` should be prevented
-     * (default: `false`)
-     */
-    disableOutsideScroll?: boolean;
+  return (
+    <MenuContentImpl
+      {...props}
+      ref={composedRefs}
+      onDismiss={() => context.onOpenChange(false)}
+      onEntryFocus={(event) => event.preventDefault()}
+      onOpenAutoFocus={composeEventHandlers(props.onOpenAutoFocus, (event) => {
+        // explicitly focus the content area only when opening
+        event.preventDefault();
+        ref.current?.focus();
+      })}
+    />
+  );
+}) as MenuRootContentPrimitive;
 
-    /**
-     * The direction of navigation between menu items.
-     * @defaultValue ltr
-     */
-    dir?: RovingFocusGroupOwnProps['dir'];
+/* ---------------------------------------------------------------------------------------------- */
 
-    /**
-     * Whether keyboard navigation should loop around
-     * @defaultValue false
-     */
-    loop?: RovingFocusGroupOwnProps['loop'];
+type MenuSubContentOwnProps = Omit<
+  Polymorphic.OwnProps<typeof MenuContentImpl>,
+  | keyof MenuContentImplPrivateProps
+  | 'side'
+  | 'portalled'
+  | 'disabledOutsidePointerEvents'
+  | 'disableOutsideScroll'
+  | 'trapFocus'
+  | 'onOpenAutoFocus'
+  | 'onCloseAutoFocus'
+>;
 
-    /**
-     * Whether the `MenuContent` should render in a `Portal`
-     * (default: `true`)
-     */
-    portalled?: boolean;
-  }
+type MenuSubContentPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof MenuContentImpl>,
+  MenuSubContentOwnProps
+>;
+
+const MenuSubContent = React.forwardRef((props, forwardedRef) => {
+  const context = useMenuContext(CONTENT_NAME);
+  const ref = React.useRef<MenuContentElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  return context.isSubmenu ? (
+    <MenuContentImpl
+      align="start"
```

Comment:
This threw me off a bit. If it's just to set a default value, we would normally do:
`const { align = 'start', ...subContentProps } = props` no?

Reply:
oh yeah..... hmmm... do we think consumers should be able to override this one?

Reply:
it's not the conventional positioning but I guess we could, although we could just simplify things by removing the prop from sub-menus until people ask for something custom, I'd be happy with that.
Especially considering that it might go away from the the root dropdown menu too once we do a more custom positioning for menus.

Reply:
Ah, it is actually already removed from the types haha... so i'll move this below the prop spread